src/ui/flutter_app/lib/widgets/drawer_user_controller.dart

```diff
@@ -58,53 +58,60 @@ class _DrawerUserControllerState extends State<DrawerUserController>
   @override
   void initState() {
     animationController = AnimationController(
-        duration: const Duration(milliseconds: 2000), vsync: this);
+      duration: const Duration(milliseconds: 2000),
+      vsync: this,
+    );
 
-    iconAnimationController = AnimationController(
-        vsync: this, duration: const Duration(milliseconds: 0));
+    iconAnimationController =
+        AnimationController(vsync: this, duration: Duration.zero);
 
-    iconAnimationController
-      ..animateTo(1.0,
-          duration: const Duration(milliseconds: 0),
-          curve: Curves.fastOutSlowIn);
+    iconAnimationController.animateTo(
+      1.0,
+      duration: Duration.zero,
+      curve: Curves.fastOutSlowIn,
+    );
 
     scrollController =
         ScrollController(initialScrollOffset: widget.drawerWidth);
 
-    scrollController
-      ..addListener(() {
-        if (scrollController.offset <= 0) {
-          if (scrollOffset != 1.0) {
-            setState(() {
-              scrollOffset = 1.0;
-              try {
-                widget.drawerIsOpen!(true);
-              } catch (_) {}
-            });
-          }
-          iconAnimationController.animateTo(0.0,
-              duration: const Duration(milliseconds: 0),
-              curve: Curves.fastOutSlowIn);
-        } else if (scrollController.offset > 0 &&
-            scrollController.offset < widget.drawerWidth.floor()) {
-          iconAnimationController.animateTo(
-              (scrollController.offset * 100 / (widget.drawerWidth)) / 100,
-              duration: const Duration(milliseconds: 0),
-              curve: Curves.fastOutSlowIn);
-        } else {
-          if (scrollOffset != 0.0) {
-            setState(() {
-              scrollOffset = 0.0;
-              try {
-                widget.drawerIsOpen!(false);
-              } catch (_) {}
-            });
-          }
-          iconAnimationController.animateTo(1.0,
-              duration: const Duration(milliseconds: 0),
-              curve: Curves.fastOutSlowIn);
+    scrollController.addListener(() {
```

Comment:
Here it feels like your own refactoring. Not sure why is the form of  `..` bad style?